File: ParsingScripts/resume_parser.py
import spacy
import sys
import re
from SkillsList.skills import skills_keywords 


nlp = spacy.load('en_core_web_sm')

def extract_name(resume_text):
    doc = nlp(resume_text)
    for ent in doc.ents:
        if ent.label_ == 'PERSON':
            return ent.text
    return ''

import re

def extract_skills(resume_text):
    skills = []
    unique_skills = set()

    # Convert the resume text to lowercase for case-insensitive matching
    resume_text_lower = resume_text.lower()

    for keyword in skills_keywords:
        keyword_lower = keyword.lower()
        if keyword_lower in resume_text_lower:
            unique_skills.add(keyword_lower)

    return list(unique_skills)


def extract_email(resume_text):
    regex = r'[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}'
    matches = re.findall(regex, resume_text)
    return matches[0] if matches else ''

def extract_experience(resume_text):
    experience_keywords = ['experience', 'employment history', 'work experience', 'professional experience', 'profile']
    experience_section = ''

    for keyword in experience_keywords:
        start_index = resume_text.lower().find(keyword)
        if start_index != -1:
            experience_section = resume_text[start_index + len(keyword):]
            break

    education_index = experience_section.lower().find('education')
    if education_index != -1:
        experience_section = experience_section[:education_index]

    # Remove leading and trailing whitespace
    experience_section = experience_section.strip()

    return experience_section


def extract_education(resume_text):
    education_keywords = ['education', 'academic qualifications']
    education_section = ''

    for keyword in education_keywords:
        start_index = resume_text.lower().find(keyword)
        if start_index != -1:
            education_section = resume_text[start_index + len(keyword):]
            break

    # Remove leading and trailing whitespace
    education_section = education_section.strip()

    return education_section



if __name__ == '__main__':
    resume_text = sys.argv[1]
    name = extract_name(resume_text)
    skills = extract_skills(resume_text)
    email = extract_email(resume_text)
    experience = extract_experience(resume_text)
    education = extract_education(resume_text)

    # Print the extracted data
    print(name)
    print(email)
    print(experience)
    print(education)

    # Print the skills
    for skill in skills:
        print(skill)

File: ParsingScripts/resume_parser.js
const natural = require('natural');
const skills_keywords = require('./skillsList/skills');

  
  const experience_keywords = ['experience', 'employment history', 'work experience', 'professional experience' , 'profile'];
  const education_keywords = ['education', 'academic qualifications'];
  
  function extractName(resume_text) {
    const tokenizer = new natural.WordTokenizer();
    const words = tokenizer.tokenize(resume_text);
    return words.slice(0, 2).join(' ');  // Assumes the first two words are the name
  }
  
  function extractSkills(resume_text) {
    const skills = [];
    const uniqueSkills = new Set();
  
    const resume_text_lower = resume_text.toLowerCase();
  
    for (const keyword of skills_keywords) {
      const keyword_lower = keyword.toLowerCase();
      if (resume_text_lower.includes(keyword_lower)) {
        uniqueSkills.add(keyword_lower);
      }
    }
  
    return Array.from(uniqueSkills);
  }
  
  
  
  
  
  
  
  
  function extractEmail(resume_text) {
    const regex = /[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/g;
    const matches = resume_text.match(regex);
    return matches ? matches[0] : '';
  }
  
  function extractExperience(resume_text) {
    let experience_section = '';
  
    for (const keyword of experience_keywords) {
      const regex = new RegExp('\\b' + keyword + '\\b', 'i');
      const match = regex.exec(resume_text);
      if (match) {
        experience_section = resume_text.substring(match.index + keyword.length);
        break;
      }
    }
  
    const education_index = experience_section.toLowerCase().indexOf('education');
    if (education_index !== -1) {
      experience_section = experience_section.substring(0, education_index);
    }
  
    return experience_section.trim();
  }
  
  function extractEducation(resume_text) {
    let education_section = '';
  
    for (const keyword of education_keywords) {
      const regex = new RegExp('\\b' + keyword + '\\b', 'i');
      const match = regex.exec(resume_text);
      if (match) {
        education_section = resume_text.substring(match.index + keyword.length);
        break;
      }
    }
  
    return education_section.trim();
  }
  
  module.exports = {
    extractName,
    extractSkills,
    extractEmail,
    extractExperience,
    extractEducation
  };
  